src/screens/ProfileScreen/elements/KarmaLock.tsx

```diff
@@ -0,0 +1,126 @@
+        </Text>
+        <Tooltip
+          allowChildInteraction={true}
+          isVisible={isTooltipShown}
+          placement={'bottom'}
+          backgroundColor={COLORS.black}
+          closeOnContentInteraction={false}
+          onClose={() => setIsTooltipShown(false)}
+          closeOnBackgroundInteraction={true}
+          contentStyle={{
+            borderRadius: 10,
+            padding: 16,
+            height: 'auto'
+          }}
+          content={
+            <View>
+              <Text
+                style={{
+                  fontWeight: '600',
+                  fontSize: normalize(16),
+                  lineHeight: normalize(24),
+                  marginBottom: normalize(8)
+                }}>
+                What is my Karma Score?
+              </Text>
+              <Text
+                style={{
+                  fontWeight: '400',
+                  fontSize: normalize(12),
+                  lineHeight: normalize(18),
+                  marginBottom: normalize(8),
+                  color: '#69707C'
+                }}>
+                The higher your Karma score, the higher your visibility on the platform. Get
+                rewarded for positive contributions to the community - and avoid being blocked by
+                others! Your Karma score will be shown as a circle around your profile picture,
+                including for your anonymous posts.
+              </Text>
+              <Text
+                style={{
+                  fontWeight: '600',
+                  paddingVertical: normalize(8),
+                  fontSize: normalize(14),
+                  lineHeight: normalize(20),
+                  color: '#4782D7',
+                  width: '100%',
+                  textAlign: 'right'
+                }}
+                onPress={() => setIsTooltipShown(false)}>
+                Got it!
+              </Text>
+            </View>
+          }>
+          <TouchableOpacity onPress={() => setIsTooltipShown(true)}>
+            <View>
+              <MemoIcQuestionMark width={normalize(13)} height={normalize(13)} />
+            </View>
+          </TouchableOpacity>
+        </Tooltip>
+      </View>
+      <TouchableOpacity
+        onPress={props.onPressCreatePost}
+        style={{
+          backgroundColor: COLORS.signed_primary,
+          height: 34,
+          marginTop: 8,
+          borderRadius: 8
+        }}>
+        <View
+          style={{
+            display: 'flex',
+            flexDirection: 'row',
+            justifyContent: 'center',
+            alignItems: 'center',
+            marginVertical: 8,
+            marginHorizontal: 10
+          }}>
+          <PencilIcon />
+          <Text
+            style={{
+              fontSize: 12,
+              color: COLORS.white,
+              marginLeft: 8
+            }}>
+            Start Posting
+          </Text>
+        </View>
+      </TouchableOpacity>
+    </View>
+  );
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `KarmaLock` component seems to be well implemented. However, it's recommended to move the inline styles into a StyleSheet object for better performance and maintainability. Inline styles in React Native can cause unnecessary re-rendering of components as a new style object is created every time the component renders. Using StyleSheet.create ensures that the styles are only created once and reused across multiple renders.

Here's an example of how you could refactor your code:

```diff
+ import {StyleSheet} from 'react-native';

...

- <View
-   style={{
-     backgroundColor: COLORS.gray100,
-     borderRadius: 12,
-     paddingVertical: 8,
-     paddingHorizontal: 4
-   }}>
+ <View style={styles.container}>

...

- <View
-   style={{
-     display: 'flex',
-     flexDirection: 'row',
-     justifyContent: 'space-between',
-     alignItems: 'center'
-   }}>
+ <View style={styles.row}>

...

- <Text
-   style={{
-     fontSize: 12,
-     color: COLORS.signed_primary,
-     fontWeight: '500',
-     paddingHorizontal: 8
-   }}>
+ <Text style={styles.text}>

...

+ const styles = StyleSheet.create({
+   container: {
+     backgroundColor: COLORS.gray100,
+     borderRadius: 12,
+     paddingVertical: 8,
+     paddingHorizontal: 4
+   },
+   row: {
+     display: 'flex',
+     flexDirection: 'row',
+     justifyContent: 'space-between',
+     alignItems: 'center'
+   },
+   text: {
+     fontSize: 12,
+     color: COLORS.signed_primary,
+     fontWeight: '500',
+     paddingHorizontal: 8
+   },
+   // ... other styles
+ });
```

This way, you can keep your render method cleaner and more readable, and it's easier to manage and update styles as they are all in one place.


<!-- This is an auto-generated comment by OSS CodeRabbit -->